src/compiler/checker.ts

```diff
@@ -22965,6 +22970,10 @@ namespace ts {
             diagnostics.add(resultDiagnostic);
         }
 
+        function isEmptyDomInterface(type: Type): boolean {
+            return isEmptyObjectType(type) && /\bHTML\w*Element\b(?![:"])/.test(typeToString(type));

```

Comment:
1. can you rewrite this to not use a regex?
2. can you write this to not use `typeToString`? Use type.symbol.escapedName instead.
3. why does the regex expect the matched string to end with ':' or '"' ? Or am I misreading it?